### ISSUE TITLE
undefined chr mutation bug 

### ISSUE THREAD

**Issue by reimand0** [reimandlab/ActiveDriverDB#131]
searching for chr17 7673776 C T

https://activedriverdb.org/protein/show/undefined

**Comment by reimand0** [comment]
I am trying to create an example of this chr>protein mapping and started from TP53 R282W. It should be "chr17 7577094 C T" according to this site, https://www.ncbi.nlm.nih.gov/clinvar/variation/12364/

We need it for two things, first as an example on front page, and second, for the manuscript.

**Comment by krassowski** [comment]
This is a strand issue. This query works: chr17 7577094 G A

**Comment by krassowski** [comment]
The corresponding rows in annotations file are:

```
chr17     7577094       g       a       TP53:NM_001126115:exon4:c.C448T:p.R150W,TP53:NM_001276761:exon8:c.C727T:p.R243W,TP53:NM_001126113:exon8:c.C844T:p.R282W,TP53:NM_001276695:exon8:c.C727T:p.R243W,TP53:NM_001126112:exon8:c.C844T:p.R282W,TP53:NM_001276760:exon8:c.C727T:p.R243W,TP53:NM_001126116:exon4:c.C448T:p.R150W,TP53:NM_001276696:exon8:c.C727T:p.R243W,TP53:NM_001276698:exon4:c.C367T:p.R123W,TP53:NM_001126114:exon8:c.C844T:p.R282W,TP53:NM_001276697:exon4:c.C367T:p.R123W,TP53:NM_001276699:exon4:c.C367T:p.R123W,TP53:NM_000546:exon8:c.C844T:p.R282W,TP53:NM_001126118:exon7:c.C727T:p.R243W,TP53:NM_001126117:exon4:c.C448T:p.R150W,
chr17     7577094       g       a       TP53:NM_001126115:exon4:c.C448T:p.R150W,TP53:NM_001276761:exon8:c.C727T:p.R243W,TP53:NM_001126113:exon8:c.C844T:p.R282W,TP53:NM_001276695:exon8:c.C727T:p.R243W,TP53:NM_001126112:exon8:c.C844T:p.R282W,TP53:NM_001276760:exon8:c.C727T:p.R243W,TP53:NM_001126116:exon4:c.C448T:p.R150W,TP53:NM_001276696:exon8:c.C727T:p.R243W,TP53:NM_001276698:exon4:c.C367T:p.R123W,TP53:NM_001126114:exon8:c.C844T:p.R282W,TP53:NM_001276697:exon4:c.C367T:p.R123W,TP53:NM_001276699:exon4:c.C367T:p.R123W,TP53:NM_000546:exon8:c.C844T:p.R282W,TP53:NM_001126118:exon7:c.C727T:p.R243W,TP53:NM_001126117:exon4:c.C448T:p.R150W,
chr17     7577094       g       a       TP53:NM_001276697:exon4:c.C367T:p.R123W,TP53:NM_001276698:exon4:c.C367T:p.R123W,TP53:NM_000546:exon8:c.C844T:p.R282W,TP53:NM_001126118:exon7:c.C727T:p.R243W,TP53:NM_001126116:exon4:c.C448T:p.R150W,TP53:NM_001126115:exon4:c.C448T:p.R150W,TP53:NM_001126114:exon8:c.C844T:p.R282W,TP53:NM_001126117:exon4:c.C448T:p.R150W,TP53:NM_001276695:exon8:c.C727T:p.R243W,TP53:NM_001276696:exon8:c.C727T:p.R243W,TP53:NM_001276699:exon4:c.C367T:p.R123W,TP53:NM_001276761:exon8:c.C727T:p.R243W,TP53:NM_001126113:exon8:c.C844T:p.R282W,TP53:NM_001276760:exon8:c.C727T:p.R243W,TP53:NM_001126112:exon8:c.C844T:p.R282W,
chr17     7577094       g       a       TP53:NM_001276761:exon8:c.C727T:p.R243W,TP53:NM_000546:exon8:c.C844T:p.R282W,TP53:NM_001126112:exon8:c.C844T:p.R282W,TP53:NM_001276695:exon8:c.C727T:p.R243W,TP53:NM_001276697:exon4:c.C367T:p.R123W,TP53:NM_001276696:exon8:c.C727T:p.R243W,TP53:NM_001126116:exon4:c.C448T:p.R150W,TP53:NM_001126113:exon8:c.C844T:p.R282W,TP53:NM_001126117:exon4:c.C448T:p.R150W,TP53:NM_001276760:exon8:c.C727T:p.R243W,TP53:NM_001126115:exon4:c.C448T:p.R150W,TP53:NM_001126118:exon7:c.C727T:p.R243W,TP53:NM_001276698:exon4:c.C367T:p.R123W,TP53:NM_001126114:exon8:c.C844T:p.R282W,TP53:NM_001276699:exon4:c.C367T:p.R123W,
```

Unfortunately there is no strand information here. We have strand data for genes so I can re-import mappings using complementary nucleotides for -1 strand. This might be easier than recreating annotations. Should I proceed?

**Comment by krassowski** [comment]
I started re-import of mappings using nucleotides from cDNA. It will take about 16 hours to complete.

**Comment by reimand0** [comment]
One option is to just offer 'did you mean XX' if the provided query is clearly on the opposite strand. What do you think?

Jüri

**Comment by krassowski** [comment]
Yes I though about this too.

**Comment by krassowski** [comment]
Import completed. The "chr17 7577094 C T" works now.

**Comment by krassowski** [comment]
Now both "chr17 7577094 C T" and "chr17 7577094 G A" works with the latter showing that the result came from "complement of chr17 7577094 G A"

**Comment by reimand0** [comment]
Thanks! Works for me as well. Two small requests remain:

1. When no mutations are present, we receive a 404 error on this page -
https://activedriverdb.org/protein/show/undefined
2. We should add a chromosome-level example onto the main page under the
search bar

**Comment by krassowski** [comment]
1. Fixed with 63125f68b84b04cbed576777be202c7de807c256.
2. Done.

I expanded placeholder text in the front page search bar to indicate that we accept genomic mutations in hg19/GRCh37 coordinates. Also I replaced commas with "or" to visually separate the examples and added "disease mutation in gene" example to indicate that the input accepts more than genes and mutations. I do not plan to add more examples into placeholder as it is already quite long.
